src/smallmap_gui.cpp

```diff
@@ -556,28 +556,32 @@ static inline uint32 GetSmallMapVegetationPixels(TileIndex tile, TileType t)
 /**
  * Return the colour a tile would be displayed with in the small map in mode "Owner".
  *
- * @param tile The tile of which we would like to get the colour.
- * @param t    Effective tile type of the tile (see #SmallMapWindow::GetTileColours).
+ * @note If include_heightmap is IH_NEVER, the return value can safely be used as a palette colour (by masking it to a uint8)
+ * @param tile              The tile of which we would like to get the colour.
+ * @param t                 Effective tile type of the tile (see #SmallMapWindow::GetTileColours).
+ * @param include_heightmap Whether to return the heightmap/contour colour of this tile (instead of the default land tile colour)
  * @return The colour of tile in the small map in mode "Owner"
  */
-static inline uint32 GetSmallMapOwnerPixels(TileIndex tile, TileType t)
+inline uint32 GetSmallMapOwnerPixels(TileIndex tile, TileType t, IncludeHeightmap include_heightmap)
 {
 	Owner o;
 
 	switch (t) {
+		case MP_VOID:     return MKCOLOUR_XXXX(PC_BLACK);
 		case MP_INDUSTRY: return MKCOLOUR_XXXX(PC_DARK_GREY);
 		case MP_HOUSE:    return MKCOLOUR_XXXX(PC_DARK_RED);
+		case MP_WATER:    return MKCOLOUR_XXXX(PC_WATER);
```

Comment:
Sorry, but this is a regression. Canals, ship depots and locks can be company-owned and are `MP_WATER` tiles. With this change, they are not visible on the smallmap anymore.

Reply:
Ok, I will revert this part of the owner check.